chaos-monkey-spring-boot/src/main/java/de/codecentric/spring/boot/chaos/monkey/watcher/outgoing/ChaosMonkeyWebClientWatcher.java

```diff
@@ -0,0 +1,75 @@
+package de.codecentric.spring.boot.chaos.monkey.watcher.outgoing;
+
+import de.codecentric.spring.boot.chaos.monkey.component.ChaosMonkeyRequestScope;
+import de.codecentric.spring.boot.chaos.monkey.component.ChaosTarget;
+import de.codecentric.spring.boot.chaos.monkey.configuration.AssaultProperties;
+import de.codecentric.spring.boot.chaos.monkey.configuration.WatcherProperties;
+import java.util.Random;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.reactive.function.client.ClientRequest;
+import org.springframework.web.reactive.function.client.ClientResponse;
+import org.springframework.web.reactive.function.client.ExchangeFilterFunction;
+import org.springframework.web.reactive.function.client.ExchangeFunction;
+import reactor.core.publisher.Mono;
+
+/** @author Marcel Becker */
+public class ChaosMonkeyWebClientWatcher implements ExchangeFilterFunction {
+
+  private final ChaosMonkeyRequestScope chaosMonkeyRequestScope;
+  private final WatcherProperties watcherProperties;
+  private final AssaultProperties assaultProperties;
+
+  public ChaosMonkeyWebClientWatcher(
+      final ChaosMonkeyRequestScope chaosMonkeyRequestScope,
+      final WatcherProperties watcherProperties,
+      AssaultProperties assaultProperties) {
+    this.chaosMonkeyRequestScope = chaosMonkeyRequestScope;
+    this.watcherProperties = watcherProperties;
+    this.assaultProperties = assaultProperties;
+  }
+
+  @Override
+  public Mono<ClientResponse> filter(
+      ClientRequest clientRequest, ExchangeFunction exchangeFunction) {
+    Mono<ClientResponse> response;
+    response = exchangeFunction.exchange(clientRequest);
+    if (watcherProperties.isWebClient()) {
+      try {
+        chaosMonkeyRequestScope.callChaosMonkey(
+            ChaosTarget.WEB_CLIENT, this.getClass().getSimpleName());
+      } catch (final Exception exception) {
+        try {
+          if (exception.getClass().equals(assaultProperties.getException().getExceptionClass())) {
+            response = Mono.just(ErrorClientResponse.getResponse());
```

Comment:
In addition to error responses, is there support for delays? Here is a snippet on how WebClient delay is done:

```
            if(latency > 0) {
                log.warn { "Chaos enabled for ${clientRequest.url()} adding latency of ${latency}ms" }
                return next.exchange(clientRequest).delayElement(Duration.ofMillis(latency))
            }
```

Reply:
I was thinking about this as well, but I had d sync upfront with @WtfJoke and @fletchgqc and we came to the conclusion, that the assaults should not be included in the watchers. The latency assault is done via  `ChaosMonkeyRequestScope`. 
The handling of the exception is just for the case, that anything else in the filter chain throws an exception except the 'ChaosMonkeyRequestScope'.

Reply:
I'm not sure I understood. Are you saying there isn't a way to leverage the assault config in this location? Or are you saying we don't want to mix the watcher and assault concerns?

I'm a bit new to the project and haven't internalized the difference between watcher and assault.

Reply:
Hi,
correct, the current status quo is not to mix assault and watchers. @WtfJoke correct me if I am wrong.